### PR TITLE
Fixes language tabs scale issue

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description
Fixes language tabs scale issue

### Proposed changes
- puts version number into the box
- limits language tabs to a max of 5, remaining languages appear in dropdown

Unbenannt
grafik
grafik

Fixes: #1030

### CODE REVIEW

File: integreat_cms/cms/templates/content_language_switcher.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<!--generic template for language switcher-->
```
Comment (timoludwig) [inline integreat_cms/cms/templates/content_language_switcher.html:1]:
```suggestion
<!--generic template for language switcher in content forms -->
```
I'd also rename this file to `content_form_language_switcher.html` to point out that's intended to be used in forms.

File: integreat_cms/cms/templates/pages/page_form.html [hunk 2]
```diff
@@ -123,24 +129,35 @@ <h1 class="heading">
...
                     </li>
                 {% endfor %}
-                {% if page_translation_form.instance.id %}
-                    <li class="self-start ml-auto px-2">
-                        {% trans 'Version' %}:
-                        {{ page_translation_form.instance.version }}
-                        (<a href="{% url 'page_revisions' page_id=page.id region_slug=region.slug language_slug=language.slug %}"
-                            class="text-blue-500 hover:underline">{% trans 'Show versions' %}</a>)
-                    </li>
+                {% if page %}
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:145]:
```suggestion
                {% if languages|length > 4 %}
```
I'd only show this language switcher if there are more than 4 languages in total.

File: integreat_cms/cms/templates/pages/page_form.html [hunk 3]
```diff
@@ -123,24 +129,35 @@ <h1 class="heading">
...
+                <div class="ml-auto z-10">
+                    {% include "content_language_switcher.html" with target='edit_page' content_id=page.id %}
+                </div>
                 {% endif %}
             </ul>
-            <div class="w-full mb-4 rounded border-2 border-blue-500 bg-white flex-auto">
+            <div class="w-full mb-4rounded border-2 border-blue-500 bg-white flex-auto">
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:151]:
```suggestion
            <div class="w-full mb-4 rounded border-2 border-blue-500 bg-white flex-auto">
```
This is obviously a typo.

File: integreat_cms/cms/templates/pages/page_form.html [hunk 4]
```diff
@@ -56,12 +56,12 @@ <h1 class="heading">
         </div>
         <div class="flex flex-wrap flex-col flex-grow pr-2 w-2/3">
             <ul class="flex flex-wrap items-end pl-4">
-                {% for other_language in languages %}
+                {% for other_language in languages|sort_languages:language|slice:":4"%}
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:59]:
```suggestion
                {% for other_language in languages|sort_languages:language|slice:":4" %}
```

File: integreat_cms/cms/templates/pages/page_form.html [hunk 5]
```diff
@@ -56,12 +56,12 @@ <h1 class="heading">
...
                     <li class="mr-1 {% if other_language == language %}z-10{% endif %}" style="margin-bottom: -2px">
                         <div class="bg-white text-blue-500 {% if other_language != language %}hover:bg-blue-500 hover:text-white{% endif %} border-l-2 border-t-2 border-r-2 border-blue-500 font-bold rounded-t-lg">
                             <div class="border-b-2 border-white">
                                 {% if other_language == language %}
-                                    <div class="{{ language.slug }} px-4">
+                                    <div class="{{ language.slug }} pt-2 pb-3 px-4">
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:64]:
```suggestion
                                    <div class="{{ language.slug }} px-4">
```
We want to keep the language tabs as narrow as they are now, this was explicitly required by the municipality team.

File: integreat_cms/cms/templates/pages/page_form.html [hunk 6]
```diff
@@ -102,7 +108,7 @@ <h1 class="heading">
                                         {{ other_language.translated_name }}
                                     </div>
                         {% else %}
-                            <a class="px-4 block" style="color: inherit;" href="{% url 'edit_page' page_id=page.id region_slug=region.slug language_slug=other_language.slug %}">
+                            <a class="px-4 block pt-2 pb-3" style="color: inherit;" href="{% url 'edit_page' page_id=page.id region_slug=region.slug language_slug=other_language.slug %}">
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:111]:
```suggestion
                            <a class="px-4 block" style="color: inherit;" href="{% url 'edit_page' page_id=page.id region_slug=region.slug language_slug=other_language.slug %}">
```

File: integreat_cms/cms/templates/pages/page_form.html [hunk 7]
```diff
@@ -123,24 +129,35 @@ <h1 class="heading">
                                         <i data-feather="x"></i>
                                     </span>
                                 {% endif %}
+                                <span>
+                                    <span class="fp fp-rounded fp-{{ other_language.primary_country_code }}"></span>
+                                    {% if other_language.secondary_country_code %}
+                                        <span class="fp fp-rounded fp-{{ other_language.secondary_country_code }}"></span>
+                                    {% endif %}
+                                </span>
                                 {{ other_language.translated_name }}
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:138]:
```suggestion
                                {{ other_language.translated_name }}
                                <span class="fp fp-rounded fp-{{ other_language.primary_country_code }}"></span>
                                {% if other_language.secondary_country_code %}
                                    <span class="fp fp-rounded fp-{{ other_language.secondary_country_code }}"></span>
                                {% endif %}
```
I think having the flags after the language name looks better.

File: integreat_cms/cms/templates/pages/page_form.html [hunk 8]
```diff
@@ -82,6 +82,12 @@ <h1 class="heading">
                                                     <i data-feather="x"></i>
                                                 </span>
                                             {% endif %}
+                                            <span>
+                                                <span class="fp fp-rounded fp-{{ other_language.primary_country_code }} ml-2"></span>
+                                                {% if other_language.secondary_country_code %}
+                                                    <span class="fp fp-rounded fp-{{ other_language.secondary_country_code }}"></span>
+                                                {% endif %}
+                                            </span>
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:90]:
```suggestion
```

File: integreat_cms/cms/templates/pages/page_form.html [hunk 9]
```diff
@@ -102,7 +108,7 @@ <h1 class="heading">
                                         {{ other_language.translated_name }}
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_form.html:108]:
```suggestion
                                        {{ other_language.translated_name }}
                                        <span class="fp fp-rounded fp-{{ other_language.primary_country_code }}"></span>
                                        {% if other_language.secondary_country_code %}
                                            <span class="fp fp-rounded fp-{{ other_language.secondary_country_code }}"></span>
                                        {% endif %}
```
Please add this change to the line 96 as well.

File: integreat_cms/cms/templatetags/content_filters.py [hunk 10]
```diff
@@ -96,6 +96,28 @@ def remove(queryset, instance):
     return queryset.exclude(id=instance.id)
 
 
+@register.filter
+def sort_languages(other_languages, current_language):
+    """
+    This filter sorts languages in language tabs
+
+    :param current_language: The current language
+    :type current_language: string
```
Comment (timoludwig) [inline integreat_cms/cms/templatetags/content_filters.py:105]:
```suggestion
    :type current_language: ~integreat_cms.cms.models.languages.language.Language
```
A language is not a string.

File: integreat_cms/cms/templatetags/content_filters.py [hunk 11]
```diff
@@ -96,6 +96,28 @@ def remove(queryset, instance):
...
+
+    :param other_languages: Other languages
+    :type other_languages: list
```
Comment (timoludwig) [inline integreat_cms/cms/templatetags/content_filters.py:108]:
```suggestion
    :type other_languages: ~django.db.models.query.QuerySet [ ~integreat_cms.cms.models.languages.language.Language ]
```
At this point in time, it's a queryset and not a list.

File: integreat_cms/cms/templatetags/content_filters.py [hunk 12]
```diff
@@ -96,6 +96,28 @@ def remove(queryset, instance):
...
+
+    :return: the filtered list with the current language on the second position
+    :rtype: list
+    """
+    if other_languages[0] != current_language:
+        other_languages = list(other_languages)
```
Comment (timoludwig) [inline integreat_cms/cms/templatetags/content_filters.py:114]:
```suggestion
    other_languages = list(other_languages)
    if other_languages[0] != current_language:
```
For the sake of consistency, let's convert this to a list anyway.

File: integreat_cms/cms/templatetags/content_filters.py [hunk 13]
```diff
@@ -96,6 +96,28 @@ def remove(queryset, instance):
...
+        other_languages.remove(current_language)
+        other_languages.insert(1, current_language)
+
+        return other_languages
```
Comment (timoludwig) [inline integreat_cms/cms/templatetags/content_filters.py:118]:
```suggestion
    return other_languages
```
If you only return the languages in the `if` case, the language tabs will be invisible on the default language

File: integreat_cms/cms/templates/content_language_switcher.html [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+                {% endif %}
+                <span class="px-2 max-w-max truncate">
+                    {{ other_language.translated_name }}
+                </span>
+                <span>
+                    <span class="fp fp-rounded fp-{{ other_language.primary_country_code }} ml-2"></span>
+                    {% if other_language.secondary_country_code %}
+                        <span class="fp fp-rounded fp-{{ other_language.secondary_country_code }}"></span>
+                    {% endif %}
+                </span>
```
Comment (timoludwig) [inline integreat_cms/cms/templates/content_language_switcher.html:41]:
```suggestion
                <span class="fp fp-rounded fp-{{ other_language.primary_country_code }} ml-2"></span>
                {% if other_language.secondary_country_code %}
                    <span class="fp fp-rounded fp-{{ other_language.secondary_country_code }}"></span>
                {% endif %}
```

File: integreat_cms/cms/templates/content_language_switcher.html [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+<div id="language-switcher" class="flex flex-col z-10 text-lg">
+    <a id="language-switcher-current" href="{% url target region_slug=region.slug language_slug=language.slug page_id=content_id %}" class="font-bold flex justify-between bg-white py-2 px-4 rounded shadow hover:bg-gray-200 border-2 border-white hover:border-gray-200">
+        Weitere Sprachen
+    </a>
+    <div id="language-switcher-list" class="h-0 hidden">
+        {% for other_language in languages|sort_languages:language|slice:"4:" %}
+            <a href="{% url target region_slug=region.slug language_slug=other_language.slug page_id=content_id %}" class="block flex justify-between whitespace-nowrap {% if forloop.last %} rounded-b {% endif %} shadow px-4 py-3 text-gray-800 hover:bg-gray-200 bg-white">
+                {% if other_language in page.languages %}
+                    {% get_translation page_form.instance other_language.slug as other_page_translation %}
```
Comment (timoludwig) [inline integreat_cms/cms/templates/content_language_switcher.html:14]:
```suggestion
                    {% get_translation instance other_language.slug as other_page_translation %}
```
If you want to, you could also generalize this template a bit more to make it possible to use it in the event form and poi form as well. In this case, you would have to pass the form instance as `instance` when including this template (similar to `target`).

**Review by timoludwig — COMMENTED** [review]
Thanks a lot! :+1: :muscle: 

I have a few ideas for improvement, have a look at my comments below:

**Review by ulliholtgrave — APPROVED** [review]
I like it. Feel free to merge it :+1:

**Pull request merged**